File: WebGeo/GeoRent.Domain/Entities/Match.cs
using System;

namespace GeoRent.Domain.Entities
{
    public class Match
    {
        public Match()
        {
            idMatch = Guid.NewGuid();
        }

        public Guid idMatch { get; set; }
        public bool like { get; set; }
        public DateTime dateTime { get; set; }
        public bool diamond { get; set; }
        public virtual Residence Residence { get; set; }
        public virtual User idUser { get; set; }
    }
}

File: app/Match.php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Match extends Model
{
    protected $hidden = ['idResidence','idUser','remember_token','created_at','updated_at'];
    
    protected $primaryKey = "idMatch";

    protected $table = 'match';

    protected $fillable = ['like','dateTime','diamond'];

    public $timestamps = false;

    public function residence()
    {
        return $this->belongsTo('App\Residence','idResidence');
    } 

    public function user()
    {
        return $this->belongsTo('App\User','idUser');
    } 
}
